inc/common.php

```diff
@@ -1358,18 +1365,28 @@ function saveWikiText($id, $text, $summary, $minor = false) {
     // if the content has not been changed, no save happens (plugins may override this)
     if (!$data['contentChanged']) return;
 
-    // confirm again event data that may altered by event handlers
-    // plugin may alter the value of oldRevision to last revision to ignore extrnal edit
-    if ($data['oldRevision'] === false) {
-        // current file was not existed, no external edit occured
-        $filesize_old = 0;
-    } elseif ($data['oldRevision'] == $currentRevision) {
-        // add current external revision entry to changelog as a regular changelog entry
+    // Confirm again both event data and pagefile that may altered by event handlers
+    //
+    // Event handlers may also modify the pagefile as well as oldRevision of event data.
+    // For example, we can imagine an action plugin that provides alternative approach for
+    // handling external edits in changelog; merging early external edits into one normal
+    // edit entry instead of separating two entries of external and normal edits. 
+    // This will be achievable if the pagefile could be restored to the last revision
+    // during $event->advise_before() using attic data.
+    //
+    $tentative = false;
+    goto DETERMINE_CHANGE_TYPE;
+    MAIN:
```

Comment:
I'm a bit surprised by the use of goto's. For me it is now tracing what happens. Is this worth a goto-structure?

Reply:
@Klap-in thanks! I agree with you that goto-structure in `saveWikiText()` is less worth and not necessary. I removed the goto loop in bbb28b9ce31573c1365c4cf71d7cb376fdee993f.